File: Embassat/View/Menu/CADEMMenuViewModel.m
//
//  CADEMMenuViewModel.m
//  Embassat
//
//  Created by Joan Romano on 22/03/14.
//  Copyright (c) 2014 Crows And Dogs. All rights reserved.
//

#import "CADEMMenuViewModel.h"

@interface CADEMMenuViewModel ()

@property (nonatomic, strong) id model;

@end

@implementation CADEMMenuViewModel

#pragma mark Lifecycle

- (instancetype)init
{
	return [self initWithModel:nil];
}

- (instancetype)initWithModel:(id)model
{
	if (self = [super init])
    {
        _model = model;
    }
    
	return self;
}

- (NSInteger)numberOfItemsInSection:(NSInteger)section
{
    return [self.model count];
}

- (NSString *)titleAtIndexPath:(NSIndexPath *)indexPath
{
    return self.model[indexPath.row];
}

@end

File: Embassat/View/Menu/CADEMMenuViewModel.swift
//
//  CADEMMenuViewModel.swift
//  Embassa't
//
//  Created by Joan Romano on 13/05/15.
//  Copyright (c) 2015 Crows And Dogs. All rights reserved.
//

import Foundation

public class CADEMMenuViewModel: NSObject, CADEMViewModelCollectionDelegate {
    
    let model: Array<String>
    
    init(model : Array<String>) {
        self.model = model
    }
    
    func numberOfItemsInSection(section : Int) -> Int {
        return self.model.count
    }
    
    public func titleAtIndexPath(indexPath: NSIndexPath) -> String {
        return self.model[indexPath.row]
    }

}
